src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.h

```diff
@@ -7,10 +7,57 @@ struct EnTimeTag;
 
 typedef void (*EnTimeTagActionFunc)(struct EnTimeTag*, PlayState*);
 
-#define ENTIMETAG_GET_1F(thisx) ((thisx)->params & 0x1F)
-#define ENTIMETAG_GET_SWITCHFLAG(thisx) ((thisx)->params & 0x7F)
-#define ENTIMETAG_GET_1FE0(thisx) (((thisx)->params & 0x1FE0) >> 0x5)
-#define ENTIMETAG_GET_E000(thisx) (((thisx)->params & 0xE000) >> 0xD)
+#define TIMETAG_GET_TYPE(thisx) (((thisx)->params & 0xE000) >> 0xD)
+
+#define TIMETAG_GET_KICKOUT_TEXT_TYPE(thisx) (((thisx)->params & 0x1FE0) >> 0x5)
+#define TIMETAG_GET_KICKOUT_EXIT_INDEX(thisx) ((thisx)->params & 0x1F)
+#define TIMETAG_KICKOUT_HOUR(thisx) ((thisx)->home.rot.x)
+#define TIMETAG_KICKOUT_MINUTE(thisx) ((thisx)->home.rot.y)
+
+#define TIMETAG_GET_SOARING_SWITCHFLAG(thisx) ((thisx)->params & 0x7F)
+
+#define TIMETAG_GET_DIARY_TYPE(thisx) ((thisx)->params & 0x7F)
+#define TIMETAG_GET_DIARY_TIMER(thisx) ((thisx)->home.rot.x)
+#define TIMETAG_GET_DIARY_SONG(thisx) ((thisx)->home.rot.z)
+
+typedef enum {
+    /* 0 */ TIMETAG_KICKOUT_DOOR,
+    /* 1 */ TIMETAG_SOARING_ENGRAVING,
+    /* 2 */ TIMETAG_ROOFTOP_OATH,
+    /* 3 */ TIMETAG_DIARY,
+    /* 4 */ TIMETAG_KICKOUT_FINAL_HOURS,
+    /* 5 */ TIMETAG_KICKOUT_NO_DOOR
+} TimeTagType;
+
+typedef enum {
+    /* 0 */ DIARY_MIKAU,
+    /* 1 */ DIARY_LULU
+} DiaryType;
+
+typedef enum {
+    /* 0 */ DIARY_SONG_EVAN_PART1,
+    /* 1 */ DIARY_SONG_EVAN_PART2
+} DiarySong;
+
+typedef enum {
+    /*  0 */ KICKOUT_TEXT_TYPE_0, // I am closing the shop
```

Comment:
Not sure `TYPE` is needed. I feel like it is more about getting the actual text, vs some kind of type of text (same comment for the macro).